### PR TITLE
Wallabag: Update Wallabag to 2.3.8 and various fixes

### PR DESCRIPTION
**Pull request by publicarray**
* update Wallabag to 2.3.8
* incorrect mysql password fix
* various fixes
* migrated to generic service script

_Linked issues:_  #3113, #3400, #3054

*Build:* https://github.com/publicarray/spksrc/releases/tag/wallabag-2.3.8

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

**Comment by ymartin59** [comment]
@publicarray Thanks for contribution. In fact I understand it is important to keep MySQL support because of previous version installation upgrade, but I would find more resource-effective to deliver package with PostgreSQL support by default, connected to DSM embedded version like for nextcloud PR #3853

In fact, I dream about a generic wizard process (shell script based) where user may select its database from a list (generated from available services in DSM): embedded postgresql, mariadb, sqlite, own package postgresql 11... with backup/restore support when uninstall/install. It would make sense for many applications...

File: spk/wallabag/Makefile [hunk 1]
```diff
@@ -13,16 +13,14 @@ ADMIN_URL = /wallabag/web/
...
 WIZARDS_DIR = src/wizard/
-
-INSTALLER_SCRIPT = src/installer.sh
-SSS_SCRIPT       = src/dsm-control.sh
-
+SERVICE_SETUP = src/service-setup.sh
+STARTABLE = no
 
 INSTALL_DEP_SERVICES = apache-web mysql
 START_DEP_SERVICES = apache-web mysql
```
Comment (ymartin59) [inline spk/wallabag/Makefile:26]: I would not enforce mysql dependency for users expecting to connect to embedded postgresql

**Comment by publicarray** [comment]
Yes that is a good idea, How about we merge this PR first to fix the package (currently it's broken) and later I can work postgresql in a separate PR?

**Comment by publicarray** [comment]
FYI [SQLite support is has been dropped](https://doc.wallabag.org/en/user/faq.html#why-has-sqlite-support-been-dropped)

**Comment by ymartin59** [comment]
In fact my idea would have a "generic database support" in framework (like for service) that a package may re-use to avoid copy/paste... and sqlite may be useful for other application like ffsync

**Comment by publicarray** [comment]
@ymartin59 I understand, but how about we do this one step at a time 😅 
1. Fix this package as is
2. add ability to select postgress, sqlite and redis (for parrallel importing suppot) for this package only
3. generate the nessesary files from the framework - also checkout https://help.synology.com/developer-guide/resource_acquisition/maria_db.html

**Comment by ymartin59** [comment]
Yes I agree. I will do some testing on this PR and publish first.
I am aware about "resource acquisition" and I guess a shell-based wizard should be able to generate require file according to user's answer "but" it would be probably easier to do all operations (connection and DB creation) in a single place in pre_install step after wizard - when we expect to use our on PostgreSQL package version or SQLite when available...

**Comment by ymartin59** [comment]
@publicarray I just discover GH action does not build non-arch package and package fails to build on my system... I still do not know why

**Comment by ymartin59** [comment]
Found: ARCH should not be override in cross/wallabag

File: spk/wallabag/src/parameters.yml [hunk 2]
```diff
@@ -1,34 +1,64 @@
 parameters:
     database_driver: pdo_mysql
+    database_driver_class: ~
     database_host: 127.0.0.1
     database_port: 3306
```
Comment (ymartin59) [inline spk/wallabag/src/parameters.yml:5]: Bad luck. On my test system, I have installed both MariaDB 5 and 10... so port is 3307 now

**Comment by ymartin59** [comment]
I have published package

**Comment by emo1509** [comment]
I get following error when installing the package on my DS2018+ : Failed to install wallabag. See errors at /var/services/web/wallabag/install.log
[wallabag install.log](https://github.com/SynoCommunity/spksrc/files/5542052/wallabag.install.log)

**Comment by publicarray** [comment]
From the log you need to check your database configuration. Do you have MariaDB installed? maybe you need to reset the password and try again?

```
Checked                  Status   Recommendation
 ------------------------ -------- ----------------------------------------------------------------------------------------------------------
  PDO Driver (pdo_mysql)   OK!
  Database connection      ERROR!   Can't connect to the database: An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused
  Database version         OK!
  curl_exec                OK!
  curl_multi_init          OK!
```

**Comment by emo1509** [comment]
I'm running MariaDB10 on Synology (port 3307). Do I need to create user & database first or will it be created by the package ?

**Comment by publicarray** [comment]
Change your port to `3306` (this is the default Mariadb port but because Synology has 2 Mariadb versions they changed the port number so they can be run side by side.)

And the wallabag install script will create the database and user for you.

**Comment by emo1509** [comment]
I have now also installed MariaDB 5 on my NAS which uses by default port 3306. I also have MariaDB 10 running because already created several databases in there for other applications. If both are running and I install Wallabag package , he creates user wallabag and database name (not yet the structure) wallabag in MariaDB 10 (which runs on port 3307). If I stop MariaDB 10 and try to install the package again, I get error message "Failed to install Wallabag. Incorrect MySQL root password." I would have expected that in the second case he would try to create user and database in the MariaDB 5.

**Comment by jamallonon** [comment]
I am having the same plain HTML issue described here: https://github.com/wallabag/wallabag/issues/4061 .  What do you guys have configured for SYMFONY__ENV__DOMAIN_NAME?

**Pull request merged**